src/services/room-service.js

```diff
@@ -119,23 +119,35 @@ export default class RoomService {
   }
 
   async verifyInvitationToken({ token, user }) {
-    let roomId = null;
-    let roomName = null;
-    let roomSlug = null;
-    let isValid = false;
+    const response = {
+      roomId: null,
+      roomName: null,
+      roomSlug: null,
+      invalidInvitationReason: null
+    };
 
     const invitation = await this.roomInvitationStore.getRoomInvitationByToken(token);
-    if (invitation?.email === user.email) {
-      const room = await this.roomStore.getRoomById(invitation.roomId);
-      if (room) {
-        roomId = room._id;
-        roomName = room.name;
-        roomSlug = room.slug;
-        isValid = true;
-      }
+    if (!invitation) {
+      response.invalidInvitationReason = INVALID_ROOM_INVITATION_REASON.token;
+      return response;
+    }
+
+    if (invitation.email !== user.email) {
+      response.invalidInvitationReason = INVALID_ROOM_INVITATION_REASON.user;
+      logger.debug(`Invitation ${invitation._id} was sent to email '${invitation?.email}' but accessed by user with email '${user.email}'`);
+      return response;
+    }
+
+    const room = await this.roomStore.getRoomById(invitation.roomId);
+    if (!room) {
```

Comment:
Technically this scenario can not occur, as deleting a room also deletes the invitations.
However I will leave the check here so that we have a clear message to show the user, in case we introduce a bug in the future.

Reply:
:+1: